crates/pera-core/src/consensus_handler.rs

```diff
@@ -355,6 +356,63 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
                             .stats
                             .inc_num_user_transactions(authority_index as usize);
                     }
+
+                    // If we receive a ProofMPCOutput transaction, verify that it's valid & create a system transaction
+                    // to store its output on the blockchain, so it will be available for the initiating user.
+                    if let ConsensusTransactionKind::SignatureMPCOutput(
+                        value,
+                        session_id,
+                        sender_address,
+                    ) = &transaction.kind
+                    {
+                        info!(
+                            "Received proof mpc output from authority {:?} for session {:?}",
+                            authority_index, session_id
+                        );
+
+                        let mut signature_mpc_manager = self.epoch_store.proof_mpc_manager.get();
+                        let is_valid_transaction = match signature_mpc_manager {
+                            Some(mpc_manager) => {
+                                let signature_mpc_manager = mpc_manager.lock().await;
+                                match signature_mpc_manager.try_verify_output(value, session_id) {
+                                    Ok(is_valid) => is_valid,
+                                    Err(e) => {
+                                        error!(
+                                            "Error verifying ProofMPCOutput output from session {:?} and party {:?}: {:?}",
+                                            session_id, authority_index, e
+                                        );
+                                        false
+                                    }
+                                }
+                            }
+                            None => {
+                                // TODO (#250): Make sure that the MPC manager is initialized before MPC events emitted.
+                                error!("MPC manager was not initialized when verifying ProofMPCOutput output from session {:?}", session_id);
+                                false
+                            }
+                        };
+
+                        if is_valid_transaction {
+                            let transaction =
+                                VerifiedTransaction::new_signature_mpc_output_system_transaction(
+                                    SignatureMPCOutput {
+                                        session_id: *session_id,
+                                        sender_address: *sender_address,
```

Comment:
I think its dangerous to take `sender_address` from the "aggregator" that sent `SignatureMPCOutput`. It is a single player that sent it, and no validation on this value occurrs here. I believe that the aggregator can make other validators assign the output to a wrong owner. We should mitigate this attack.

Reply:
I agree, there should be a check that the `sender_address` is the same as the address of the authority who created the block that contained this MPC Message.

Reply:
I think there's a confusion here; the `sender_address` is the address of the initiating user, not of the aggregator

Reply:
WIll rename it to `initiating_address`

Reply:
There was no confusion, I understood it as you intended. Please don't resolve conversations like these by yourselves. I believe it is dangerous to take this value as you do here. Please attend to this.

Reply:
@ItayLevyOfficial @yaelAbergel attend to this please

Reply:
Created an [issue](https://github.com/dwallet-labs/dwallet-network/issues/323) for that, I've added a matching TODO in that spot in the DKG pr. I'll start working on it today

Reply:
https://github.com/dwallet-labs/dwallet-network/pull/324